pages/api/id/[form]/retrieval.ts

```diff
@@ -0,0 +1,207 @@
+ */
+function connectToDynamo(): DynamoDBDocumentClient {
+  //Create dynamodb client
+  const db = new DynamoDBClient({
+    region: process.env.AWS_REGION ?? "ca-central-1",
+    endpoint: process.env.LOCAL_DYNAMODB_ENDPOINT,
+  });
+
+  return DynamoDBDocumentClient.from(db);
+}
+
+/**
+ * Helper function to retrieve up to 10 form responses from dynamoDB
+ * using the retrieved-index
+ * @param res - The NextJS response object
+ * @param formID - The relevant form ID from which to retrieve form responses
+ */
+async function getLatestResponseList(
+  res: NextApiResponse,
+  formID: string
+): Promise<ResponseListResult | boolean> {
+  try {
+    const documentClient = connectToDynamo();
+    //Create form's responses db param
+    //NB: A filter expression is applied after a Query finishes, but before the results are returned.
+    //Therefore, a Query consumes the same amount of read capacity, regardless of whether a filter expression is present.
+    const getItemsDbParams = {
+      TableName: "Vault",
+      IndexName: "retrieved-index",
+      Limit: 10,
+      //Cannot use partitin key or sort key such as formID and Retrieved attribute in keyConditionExpression simultaneously
+      ExpressionAttributeValues: {
+        ":formID": formID,
+        ":retrieved": 0,
+      },
+      KeyConditionExpression: "Retrieved = :retrieved",
+      FilterExpression: "FormID = :formID",
+      //A filter expression cannot contain partition key or sort key attributes.
+      //You need to specify those attributes in the key condition expression, not the filter expression.
+      ProjectionExpression: "FormID,SubmissionID,FormSubmission,Retrieved",
+    };
+    //Get form's responses for formID
+    return {
+      formResponses: await documentClient.send(new QueryCommand(getItemsDbParams)),
+      documentClient,
+    };
+  } catch (error) {
+    logMessage.error(error as Error);
+    res.status(500).json({ error: "Error on Server Side when fetching form's responses" });
+  }
+  return false;
+}
+
+/**
+ * Request type: GET
+ * USAGE:
+ * curl http://gc-forms/api/id/1/retrieval
+   -H "Accept: application/json"
+   -H "Authorization: Bearer {token}"
+
+ * This method will fetch up to 10 form responses and return them to the user
+ * @param req - The NextJS request object
+ * @param res - The NextJS response object
+ * @param formID - The form ID from which to retrieve responses
+ * @param email - The email of the authenticated user carrying out the request
+ * @param temporaryToken - The valid JWT temporary token of the authenticated user carrying out the request
+ */
+async function getFormResponses(
+  req: NextApiRequest,
+  res: NextApiResponse,
+  formID: string,
+  email?: string,
+  temporaryToken?: string
+): Promise<void> {
+  const result = await getLatestResponseList(res, formID);
+  // function will return false if there is an error. The response will have
+  // already been set so we do not need to continue with the function execution
```

Comment:
Not sure it is a good idea since we could basically get rid of this comment just by moving the response logic from the helper function to here.

Reply:
Think you're right here. The helper function was a remnant of my previous implementation where it was reused. I must have forgotten to move back the code